packages/frontend/src/components/common/PivotTable/index.tsx

```diff
@@ -53,63 +83,143 @@ const PivotTable: FC<PivotTableProps> = ({
     getFieldLabel,
     getField,
     className,
+    showSubtotals = false,
     ...tableProps
 }) => {
     const containerRef = useRef<HTMLDivElement>(null);
+    const [grouping, setGrouping] = React.useState<GroupingState>([]);
 
-    const getItemFromAxis = useCallback(
-        (rowIndex: number, colIndex: number) => {
-            const value = data.pivotConfig.metricsAsRows
-                ? last(data.indexValues[rowIndex])
-                : last(data.headerValues)?.[colIndex];
-
-            if (!value || !value.fieldId) throw new Error('Invalid pivot data');
-
-            return getField(value.fieldId);
-        },
-        [
-            data.pivotConfig.metricsAsRows,
-            data.headerValues,
-            data.indexValues,
-            getField,
-        ],
-    );
-
-    const getRowTotalValueFromAxis = useCallback(
-        (total: unknown, colIndex: number): ResultValue => {
-            const value = last(data.rowTotalFields)?.[colIndex];
-
-            if (!value || !value.fieldId) throw new Error('Invalid pivot data');
-            const item = getField(value.fieldId);
+    const hasColumnTotals = data.pivotConfig.columnTotals;
 
-            const formattedValue = formatItemValue(item, total);
+    const hasRowTotals = data.pivotConfig.rowTotals;
 
-            return {
-                raw: total,
-                formatted: formattedValue,
-            };
+    const { columns, columnOrder } = useMemo(() => {
+        let headerInfoForColumns = data.headerValues[0].map(() => ({}));
+        headerInfoForColumns = [
+            ...Array(data.indexValueTypes.length),
+            ...headerInfoForColumns,
+        ];
+
+        const newColumnOrder: string[] = [];
+        if (!hideRowNumbers) newColumnOrder.push(ROW_NUMBER_COLUMN_ID);
+
+        let newColumns = data.retrofitData.pivotColumnInfo.map(
+            (col, colIndex) => {
+                newColumnOrder.push(col.fieldId);
+
+                const item = getField(col.underlyingId || col.baseId);
```

Comment:
Thanks again. Changed it up a bit to leave `item` undefined if there's no `underlyingId` or `baseId` since `item` is not needed unless those IDs are available.